lightning/src/chain/channelmonitor.rs

```diff
@@ -2272,6 +2113,70 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
 	}
 }
 
+/// Persist is responsible for persisting channel monitors: this could mean
+/// writing once to disk, and/or uploading to one or more backup services.
+///
+/// Note that for every new monitor, you **must** persist the new ChannelMonitor
+/// to disk/backups. And, on every update, you **must** persist either the
+/// ChannelMonitorUpdate or the updated monitor itself. Otherwise, there is risk
+/// of situations such as revoking a transaction, then crashing before this
+/// revocation can be persisted, then unintentionally broadcasting a revoked
+/// transaction and losing money. This is a risk because previous channel states
+/// are toxic, so it's important that whatever channel state is persisted is
+/// kept up-to-date.
+///
+/// There are several different strategies for persistence, with different
+/// requirements. While not exhaustive, here are a few. A user could:
+///   * have multiple redundant backups where the user wants to ensure that
+///     every backup is always accessible. In this case, implementors of this
+///     trait need to ensure that every copy is updated at each step before
+///     moving forward, with none falling out of sync,
+///   * have backups where they're OK with relying on only M of N backups
+///     temporarily, in which case the user doesn't care about consensus between
+///     backups and only needs to ensure that they read from the backup(s) with
+///     the highest [`ChannelMonitorUpdate::update_id`].
+///
+/// [`ChannelMonitorUpdate::update_id`]: struct.ChannelMonitorUpdate.html#structfield.update_id
+pub trait Persist<Keys: ChannelKeys>: Send + Sync {
+	/// Persist a new channel's data. The data can be stored with any file
+	/// name/path, but the identifier provided is the channel's outpoint. Note
+	/// that you **must** persist every new monitor to disk. See the Persist trait
+	/// documentation for more details, and for information on consensus between
+	/// backups.
+	///
+	/// See [`ChannelMonitor::write_for_disk`] for writing out a ChannelMonitor.
+	///
+	/// [`update_id`]: struct.ChannelMonitorUpdate.html#structfield.update_id
+	/// [`ChannelMonitor::write_for_disk`]: ../../chain/channelmonitor/struct.ChannelMonitor.html#method.write_for_disk
+	fn persist_new_channel(&self, id: OutPoint, data: &ChannelMonitor<Keys>) -> Result<(), ChannelMonitorUpdateErr>;
```

Comment:
Maybe link to the docs for ChannelMonitorUpdateErr for requirements when returning errors (so that users can at least get a link telling them to call ChannelManager::monitor_update_restored or whatever that function is called).